src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -2780,11 +2780,18 @@ do_compact_test(const compact_test_args args, storage_test_fixture& f) {
     // the beginning of the log
     BOOST_REQUIRE_EQUAL(final_gaps.num_gaps, 1);
     BOOST_REQUIRE_EQUAL(final_gaps.first_gap_start, model::offset(0));
+
+    // If adjacent segment compaction worked in order from oldest to newest, we
+    // could use this assert.
+#if 0
```

Comment:
can we simply remove this if the assertion is incorrect ? also it should be squashed with previous changes to keep history bisectable

Reply:
Did you see the commit comment for this patch? I think this is useful documentation.. It took a while to figure out the behavior here. I agree on the squishing part--I pulled this out for reviewers' benefit.

Reply:
LMK if you still want the comment removed.

Reply:
this seems reasonable, but using a multi-line comment instead of an if 0 block would be more idiomatic in the code base. this particular case seems very clear and the if 0/endif kind of highlights the aspect of it being code that the comment is referring to

Reply:
`#if false` feels like a merge-broken version check or something, agree with Noah, it's better to use a multi-line comment